commit ae1fad5bb773c4c58c2249472b5c06ca05ebe702
Author: Christoph M. Becker <cmbecker69@gmx.de>
Date:   Wed Jan 9 14:26:18 2019 +0100

    Fix #77431 SplFileInfo::__construct() accepts NUL bytes
    
    `SplFileInfo::__construct()` has to expect a path instead of a string,
    analogous to `SplFileObject::__construct()`.

diff --git a/ext/spl/spl_directory.c b/ext/spl/spl_directory.c
index ef2e72d36c..e690b94615 100644
--- a/ext/spl/spl_directory.c
+++ b/ext/spl/spl_directory.c
@@ -1131,7 +1131,7 @@ SPL_METHOD(SplFileInfo, __construct)
 	char *path;
 	size_t len;
 
-	if (zend_parse_parameters_throw(ZEND_NUM_ARGS(), "s", &path, &len) == FAILURE) {
+	if (zend_parse_parameters_throw(ZEND_NUM_ARGS(), "p", &path, &len) == FAILURE) {
 		return;
 	}
 
diff --git a/ext/spl/tests/bug77431.phpt b/ext/spl/tests/bug77431.phpt
new file mode 100644
index 0000000000..eb1ca96b75
--- /dev/null
+++ b/ext/spl/tests/bug77431.phpt
@@ -0,0 +1,9 @@
+--TEST--
+Bug #77431 (SplFileInfo::__construct() accepts NUL bytes)
+--FILE--
+<?php
+new SplFileInfo("bad\0good");
+?>
+--EXPECTF--
+Fatal error: Uncaught TypeError: SplFileInfo::__construct() expects parameter 1 to be a valid path, string given in %s:%d
+Stack trace:%A
\ No newline at end of file
